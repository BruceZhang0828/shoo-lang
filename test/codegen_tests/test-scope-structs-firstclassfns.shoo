// struct of same name inside firstclass fn inside struct of same name
struct Fancy {
	int field;
	string hello;
	func(int;string) anon = function (int x) string {
		struct Fancy {
			string diff;
			string diff2;
			int storeStuff;
		}
		Fancy innerFancy = new(Fancy);
		innerFancy.diff = "secret message is here";
		innerFancy.diff2 = "second secret message";

		return innerFancy.diff + innerFancy.diff2 + str_of_int(x);
	};
}

Fancy myNew;
myNew = new(Fancy);

myNew.hello = "HI";
string result = myNew.anon(6);
println(myNew.hello);
println(result);