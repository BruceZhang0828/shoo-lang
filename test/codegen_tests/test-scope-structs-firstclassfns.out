HI
secret message is heresecond secret message6